src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -320,7 +320,10 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
     stx.tx.receivingAddress match {
       case None =>
         val address = worldStateProxy.createAddress(stx.senderAddress)
-        val world1 = worldStateProxy.newEmptyAccount(address)
+        val world1 =
+          // YP eq (86) implies that the address might have already received funds before contract init. We need to check that.
+          if(worldStateProxy.getAccount(address).isDefined) worldStateProxy
+          else worldStateProxy.newEmptyAccount(address)
```

Comment:
Why do we even bother with this step? `WorldStateProxy#transfer` is already an upsert. Look at the implementation in `CREATE` opcode.